### ISSUE TITLE
Webman Offline mode enabled fails to block all PSN and thirdparty servers

### ISSUE THREAD

**Issue by 444nonymous** [aldostools/webMAN-MOD#8]
https://github.com/aldostools/webMAN-MOD/issues/5

Sorry to insist, like reported here (but closed, so I cannot reply), the PS3 detects an internet connection and tries to connect to PSN again and again. It also tries to connect and login to third party servers used by EA, Ubisoft, Square, etc.

Since it's hard to know all the servers servers each games uses, and the tracking consequences it has every time it tries to connect, I suggest implementing a more restrictive and real offline mode, allowing only the private ip range when playing games.

This would solve both privacy issues and annoying PSN login prompts. 

Thank you for considering it.

**Comment by aldostools** [comment]
webMAN MOD already has the option to block the known PSN servers. You already answered yourself: "it's hard to know all the servers servers each games uses". 

The issue that you're mentioning is not an issue in the plugin. It is the way the games work.
You have to patch the games or make the suggestion to SIE.

I can't do anything, unless you make a pull request with the actual code to do it or provide me the technical documentation needed to do what you suggested.

**Comment by 444nonymous** [comment]
How do you block PSN servers? Do you use some kind of black list?

For direct IP connections (not DNS resolved) block anything not matching the private range:
10.0.0.0 - 10.255.255.255
172.16.0.0 - 172.31.255.255
192.168.0.0 - 192.168.255.255

For connections to server names (DNS resolved) block any server ending with a Top-Level Domain:
http://data.iana.org/TLD/tlds-alpha-by-domain.txt

Since internet connections can only be made to specific IPs and domains, this way you could still access the local IP range and any DNS domain you may have invented for your local network. IE "router.myhome" resolves to 192.168.1.1 and "ps3.myhome" resolves to 192.168.1.9, etc.

**Comment by aldostools** [comment]
The PSN servers are blocked replacing with NULL characters, the domain names of the URLs found in LV1 and LV2. However, this is done only for URLs in GameOS. For the games you have to patch the games individually.

This is the code used to block the servers:
https://github.com/aldostools/webMAN-MOD/blob/master/include/stealth.h#L176

If you have a better method to do it, please provide me the source code and I'll be glad to include it in the project and give you the credits.

**Comment by 444nonymous** [comment]
Could it be possible to create an option to automatically turn "settings -> network -> internet connection" off when starting a game, and on again when quitting a game?

That way you would only have to change one available ps3 setting option when detecting starting and quitting game events.

**Comment by aldostools** [comment]
You can try this **_test build**_
http://aldostools.org/temp/test/webftp_server_full.sprx

It has a new 'Offline' option in /setup.ps3 
If you sets it to "Games", the plugin will disable the network status in the registry when a game starts.
When the system returns to XMB the network status is re-enabled in the registry.

If you prefer, you can set the option per game adding the tag [offline] to the file name of the ISO or folder name.

Please, report back if it's working.

**Comment by 444nonymous** [comment]
I see that webMAN MOD 1.43.35 includes that option. I've tested it with a few games and quitting to XMB first then loading other games, quitting from game to Movian directly (needs online). Turning the ps3 off from game and getting online as default when restarting, etc. It also works for PSN installed games launched from the XMB (in this case I don't know if it can be enabled / disabled as a per game option). So far all I've tried it's working fine. (I'm using latest Rebug)

**Comment by 444nonymous** [comment]
Problems:
1. Info message says "offline" when going offline and "offline" again when going back online. It should be better to indicante "online" when online.
2. If  "settings -> network -> internet connection" is set off and you clear all webman temp / cache / config files, OR if "settings -> network -> internet connection" is set off you and update webman, configuring it for the first time to block "GAME", network will always be off and never turned on by webman. When this problem happens you cannot connect to webman by FTP because the PS3 is always really offline, maybe as expected before webman had this option to switch the network.

Thanks!

**Comment by aldostools** [comment]
I did a quick fix for problem 1 (but I haven't tested it due I'll be out).
If you want, you can test it from:
http://aldostools.org/temp/test/webftp_server_full.sprx

Regarding the problem 2:
a) if the user removes the file /dev_hdd0/tmp/wm_disabled manually, there is no way that webman knows that the network was on and it should be restored. In that case, the user must enable the network manually.
b) if the network was already off, webman is not supposed to turn it on.

**Comment by 444nonymous** [comment]
I don't see any "online" message no matter what. And there is some weird behavior. 

If I launch irisman an offline message popups (select + start shows offline too) but I can download covers and updates (so it's working like it should). When I quit irisman to the XMB no message is displayed, it should be "online" (select + start shows online).

When I launch Movian it's always online (no messages). It does not matter if I launch it from XBM (already online) or from an offline game / irisman.

If I launch a game it shows "offline", but no message (it should be "online") when quitting. The old behavior was to show "offline" when launching and again "offline" when quitting.

I have also noticed that you removed this feature from webman lite, the one I use the most :(

**Comment by aldostools** [comment]
I confirmed that the "online" message is sent. But in most of the cases it happens while VSH is switching the screen mode. So it rarely is seen.

Regarding IRISMAN being able to download, it is normal. The "offline" mode if a spoof only turns off the setting in the registry, but the network is still working, The games normally read this setting first and believe that the network is disabled. IRISMAN doesn't do that check. If you navigate to the PS3 with the PC, you will notice that you also can access webMAN.

Movian and other apps (e.g. Netflix, Youtube, etc) were added to the exception list.
That list can be edited in /dev_hdd0/tmp/wm_online_ids.txt

I'll consider to re-add the feature to webman lite if it doesn't has big in the footprint.

EDIT: the lite edition now has the feature re-added.
